### PR TITLE
Remove unused props, add baseUrl into kernel Connection metadata

### PR DESCRIPTION
**Pull request by DonJayamanne**
Minor refactoring, basically include the Remote connection information (just baseUrl) in Kernel Connection metadata

### CODE REVIEW

File: src/client/datascience/jupyter/interactiveWindowDebugger.ts [hunk 1]
```diff
@@ -213,8 +211,7 @@ export class InteractiveWindowDebugger implements IInteractiveWindowDebugger, IC
         // installed locally by the extension
         // Actually until this is resolved: https://github.com/microsoft/vscode-python/issues/7615, skip adding
         // this path.
-        const connectionInfo = kernel.connection;
-        if (connectionInfo && connectionInfo.localLaunch) {
+        if (isLocalConnection(kernel.kernelConnectionMetadata)) {
```
Comment (DonJayamanne) [inline src/client/datascience/jupyter/interactiveWindowDebugger.ts:214]: This allows us to decouple the code form the `connection` information.
Getting the `connection` information is not cheap/easy.

**Review by rchiodo — APPROVED** [review]

File: src/client/datascience/jupyter/jupyterConnection.ts [hunk 2]
```diff
@@ -243,8 +241,6 @@ class JupyterConnection implements IJupyterConnection {
             childProc.on('exit', (c) => {
                 // Our code expects the exit code to be of type `number` or `undefined`.
                 const code = typeof c === 'number' ? c : 0;
-                this.valid = false;
-                this.localProcExitCode = code;
```
Comment (DonJayamanne) [inline src/client/datascience/jupyter/jupyterConnection.ts:247]: All of this was only required in `variable` viewer.
Var viewer should no longer need this, none of the other places (intellisense, kernel execution, etc rely on these).
Hence hope is to use a standard way to detect whether the process/kernel is dead

File: src/client/datascience/types.ts [hunk 3]
```diff
@@ -61,22 +61,19 @@ export interface IDisplayOptions {
...
     readonly type: 'raw';
     readonly localLaunch: true;
-    readonly valid: boolean;
     readonly displayName: string;
 }
 
 export interface IJupyterConnection extends Disposable {
     readonly type: 'jupyter';
     readonly localLaunch: boolean;
-    readonly valid: boolean;
```
Comment (DonJayamanne) [inline src/client/datascience/types.ts:71]: More stuff removed

File: src/client/datascience/jupyter/kernels/types.ts [hunk 4]
```diff
@@ -66,6 +67,7 @@ export type RemoteKernelSpecConnectionMetadata = Readonly<{
     interpreter?: undefined;
     kernelSpec: IJupyterKernelSpec;
     kind: 'startUsingRemoteKernelSpec';
+    baseUrl: string;
```
Comment (DonJayamanne) [inline src/client/datascience/jupyter/kernels/types.ts:70]: This is simpler, also allows us in the future to connect to multiple remotes or even have local + remote at the same time

**Pull request merged**